docs/products/opensearch/howto/opensearch-search-and-python.rst

```diff
@@ -0,0 +1,471 @@
+This query can be used to match phrases in a field. Where the ``query`` is the phrase that is being searched in a certain field:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "match_phrase": {
+                        field: {
+                          "query": query
+                        }
+                      }
+                    }
+                  }
+
+If we are looking for a certain phrase, for example, ``pannacotta with lemon marmalade`` in the title, we may use a query like:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "match_phrase": {
+                        "title": {
+                          "query": "pannacotta with lemon marmalade"
+                        }
+                      }
+                    }
+                  }
+    resp = client.search(index=INDEX_NAME, body=query_body)
+    log_titles(resp)
+
+If you know exactly which phrases you are looking for in a recipe, you can try out our ``match_phrase`` demo:
+
+::
+
+  python search.py match_phrase "title" "Pannacotta with lemon marmalade"
+
+If you just have a rough idea of the phrase you are looking for, you can make your match phrase query more flexible with the ``slop`` parameter in the next section.
+
+Match phrase with slop query
+----------------------------
+A useful feature we can make use of in the match_phrase query is the “slop” parameter which allows us to create more flexible searches. If we are searching for ``pannacotta marmalade`` with the ``match_phrase`` query, no results would be returned.
+We can solve this by setting the ``slop`` parameter. The ``slop`` parameter allows to control the degree where the order can be off the order, the default value is 0. The query can be constructed as:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "match_phrase": {
+                        title: {
+                          "query": query
+                          "slop": slop
+                        }
+                      }
+                    }
+                  }
+
+Suppose we are looking for ``pannacotta marmalade`` phrase. In order to find more results rather than exact phrases, we should allow a certain degree like setting the ``slop=2``, so it can find matches skipping two words between the searched ones.
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "match_phrase": {
+                        "title": {
+                          "query": "pannacotta marmalade"
+                          "slop": 2
+                        }
+                      }
+                    }
+                  }
+
+With this flexibility, we can find titles with the desired words even if there are other words in between all thanks to the ``slop`` parameter.
+
+.. code-block:: python
+
+    ['Lemon Pannacotta with Lemon Marmalade ',
```

Comment:
wait, you shouldn't get two same results, right? maybe you loaded dataset twice?

Reply:
I added the data twice. Thanks!